test/built-ins/String/prototype/Symbol.iterator/prototype/ancestry.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 21.1.5.2
+description: >
+    The [[Prototype]] internal slot ofthe %StringIteratorPrototype% is the
+    %IteratorPrototype% intrinsic object (25.1.2).
+---*/
+
+var strItrProto = Object.getPrototypeOf(String.prototype[Symbol.iterator]());
+var itrProto = Object.getPrototypeOf(
+  Object.getPrototypeOf([][Symbol.iterator]())
+);
+
+assert.sameValue(Object.getPrototypeOf(strItrProto), itrProto);
```

Comment:
This test will pass even if there is no intermediate %IteratorPrototype%. Maybe you can check the next [[Prototype]] too and make sure it is %ObjectPrototype%?